packages/ui-components/src/overlay.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+
+/**
+ * A namespace for functions and types related to creating overlays.
+ */
+export namespace Overlay {
+  /**
+   * Options for creating an overlay.
+   */
+  export interface IOptions {
+    /**
+     * The element that will host the overlay.
+     */
+    hostElement: Element;
+
+    /**
+     * Configuration for the overlay's class names and optional title.
+     */
+    classConfig: {
+      /**
+       * The primary class name to be applied to the overlay.
+       */
+      class: string;
+
+      /**
+       * An optional title to be set on the overlay element.
+       */
+      title?: string;
+    };
+
+    /**
+     * Configuration for the overlay's CSS styles.
+     */
+    cssConfig: {
+      /**
+       * The position of the overlay (e.g., "absolute", "relative", etc.).
+       */
+      position: string;
+
+      /**
+       * The top position of the overlay.
+       */
+      top: string;
+
+      /**
+       * The left position of the overlay.
+       */
+      left: string;
+
+      /**
+       * The width of the overlay.
+       */
+      width: string;
+
+      /**
+       * The height of the overlay.
+       */
+      height: string;
+
+      /**
+       * Optional border radius for the overlay.
+       */
+      'border-Radius'?: string;
+
+      /**
+       * Optional z-index for the overlay.
+       */
+      'z-index'?: string;
+
+      /**
+       * Optional text alignment for the overlay's content.
+       */
+      'text-align'?: string;
+
+      /**
+       * The text color of the overlay's content.
+       */
+      color: string;
+
+      /**
+       * Optional inner HTML content for the overlay.
+       */
+      innerHTML?: string;
+
+      /**
+       * The background color of the overlay.
+       */
+      background: string;
+
+      /**
+       * The visibility of the overlay (e.g., "visible", "hidden").
+       */
+      visibility: string;
+    };
+  }
+
+  /**
+   * Creates an overlay element and appends it to the specified host element.
+   *
+   * @param options - The options for creating the overlay.
+   */
+  export function createOverlay(options: IOptions): void {
+    const { hostElement, classConfig, cssConfig } = options;
+
+    // Check if the overlay already exists to avoid duplicates.
+    if (!hostElement.querySelector(`[title="${classConfig.title}"]`)) {
```

Comment:
Matching by tittle is brittle; moreover `querySelector` can lead to security issues if the overlay was used on user-controlled notebook UI; I would suggest that the responsibility for this is move a level higher.

Reply:
I've changed this to match on class rather than title and moved it to a higher level.